src/MicroSDC.cpp

```diff
@@ -239,7 +239,8 @@ std::string MicroSDC::calculateMessageID()
 void MicroSDC::addMdState(std::shared_ptr<StateHandler> stateHandler)
 {
   stateHandler->setMicroSDC(this);
-  stateHandlers_.emplace_back(std::move(stateHandler));
+  auto handle = stateHandler->getDescriptorHandle();
+  stateHandlers_.emplace(handle, std::move(stateHandler));
```

Comment:
Is there a reason for an additional copy here?
```suggestion
  stateHandlers_.emplace(stateHandler->getDescriptorHandle(), std::move(stateHandler));
```

Reply:
No, not really. It was meant for better readability, but I guess it can be omitted in this case.